yellowbrick/classifier/rocauc.py

```diff
@@ -168,8 +169,30 @@ def score(self, X, y=None, **kwargs):
         # Compute the predictions for the test data
         y_pred = self._get_y_scores(X)
 
-        # # Classes may be label encoded so only use what's in y to compute.
-        # # The self.classes_ attribute will be used as names for labels.
+        # Check to see if shape of predictions signals a binary decision
+        if len(y_pred.shape) == 1:
+            self._binary_decision = True
+
+            # Raise an error if it's a binary decision and user has set micro,
+            # macro, or per_class to True
+            if self.micro or self.macro or self.per_class:
+                raise ModelError(
```

Comment:
I love this exception message - it's not that there is a bug in ROCAUC, it's just that there are a set of conditions when the defaults don't work - this error explains how to fix it when that case is encountered and gives our users a clear path to success!